### PR TITLE
Clarify multiple Local/remote endpoints

### LINKED ISSUE #1339: Define "represent the same host"
From the review by Lars Eggert, COMMENT part:

### Section 6, paragraph 6
```
    If more than one Local Endpoint is specified on a Preconnection, then
    all the Local Endpoints on the Preconnection MUST represent the same
    host.  For example, they might correspond to different interfaces on
    a multi-homed host, of they might correspond to local interfaces and
    a STUN server that can be resolved to a server reflexive address for
    a Preconnection used to make a peer-to-peer Rendezvous.
```
In order to make this MUST enforceable, a concrete definition of
what "represent the same host" means is IMO needed. Is this a static
or dynamic check?

### LINKED ISSUE #1340: Programmer can't know that all endpoints are the same service
From the review by Lars Eggert, COMMENT part:

### Section 6, paragraph 6
```
    If more than one Remote Endpoint is specified on the Preconnection,
    then all the Remote Endpoints on the Preconnection should represent
    the same service, to the extent that the application and the
    Transport Services system can validate that the Remote Endpoints are
    indeed the same service.  For example, the Remote Endpoints might
    represent various network interfaces of a host, or a server reflexive
    address that can be used to reach a host, or a set of hosts that
    provide equivalent local balanced service.
```
It's a lowercase "should", but I still have no idea how a programmer
or the API would verify whether all endpoints are the same service.
Seems unenforceable.

### PR DESCRIPTION
**Pull request by philsbln**
Replace the over-specific definitions with racing compatible equivalents without mentioning racing.

Fixes #1339 
Fixes #1340

### CODE REVIEW

File: draft-ietf-taps-interface.md [hunk 1]
```diff
@@ -680,17 +680,17 @@ the Preconnection is used to `Listen` for incoming Connections.
 At least one Local Endpoint and one Remote Endpoint MUST be specified if a
 peer-to-peer `Rendezvous` is to occur based on the Preconnection.
 
-If more than one Local Endpoint is specified on a Preconnection, then all
-the Local Endpoints on the Preconnection MUST represent the same host. For
-example, their Endpoint Identifiers might correspond to different interfaces on a multi-homed
+If Local Endpoints are specified on a Preconnection, then all the Local Endpoints
+on the Preconnection MUST eliable for communication from the host they are specified on.
+For example, their Endpoint Identifiers might correspond to different interfaces on a multi-homed
```
Comment (gorryfair) [inline draft-ietf-taps-interface.md:685]: I do not understand /eliable/ ?

File: draft-ietf-taps-interface.md [hunk 2]
```diff
@@ -680,17 +680,17 @@ the Preconnection is used to `Listen` for incoming Connections.
...
 host, or their Endpoint Identifiers might correspond to local interfaces and a STUN server that
 can be resolved to a server reflexive address for a Preconnection used to
 make a peer-to-peer `Rendezvous`.
 
-If more than one Remote Endpoint is specified on the Preconnection, then
-all the Remote Endpoints on the Preconnection SHOULD represent the same
-service, to the extent that the application and the Transport Services
-system can validate that the Remote Endpoints correspond to the same service.
+If more than one Remote Endpoint is specified on the Preconnection,
+the transport system SHOULD assume that all the Remote Endpoints on the
```
Comment (mwelzl) [inline draft-ietf-taps-interface.md:691]:
```suggestion
the Transport Services system SHOULD assume that all the Remote Endpoints on the
```

**Review by mwelzl — COMMENTED** [review]
Just a nit

Else, I see... you say the system should assume it, instead of saying it should represent - makes more sense to me!  but I feel a bit insecure about the whole thing and would want others to take the go/no-go decision on this one.

File: draft-ietf-taps-interface.md [hunk 3]
```diff
@@ -680,17 +680,17 @@ the Preconnection is used to `Listen` for incoming Connections.
 At least one Local Endpoint and one Remote Endpoint MUST be specified if a
 peer-to-peer `Rendezvous` is to occur based on the Preconnection.
 
-If more than one Local Endpoint is specified on a Preconnection, then all
-the Local Endpoints on the Preconnection MUST represent the same host. For
-example, their Endpoint Identifiers might correspond to different interfaces on a multi-homed
+If Local Endpoints are specified on a Preconnection, then all the Local Endpoints
+on the Preconnection MUST be eligible for communication from the host they are specified on.
+For example, their Endpoint Identifiers might correspond to different interfaces on a multi-homed
```
Comment (tfpauly) [inline draft-ietf-taps-interface.md:685]:
```suggestion
If more than one Local Endpoint is specified on a Preconnection, then the application
 is indicating that all of the Local Endpoints are eligible to be used for Connections. For
 example, their Endpoint Identifiers might correspond to different interfaces on a multi-homed
```

File: draft-ietf-taps-interface.md [hunk 4]
```diff
@@ -680,17 +680,17 @@ the Preconnection is used to `Listen` for incoming Connections.
...
 make a peer-to-peer `Rendezvous`.
 
-If more than one Remote Endpoint is specified on the Preconnection, then
-all the Remote Endpoints on the Preconnection SHOULD represent the same
-service, to the extent that the application and the Transport Services
-system can validate that the Remote Endpoints correspond to the same service.
+If more than one Remote Endpoint is specified on the Preconnection,
+the Transport Services system SHOULD assume that all the Remote Endpoints on the
+Preconnection represent equivalent services, to the extent that the
+the Transport Services system can choose any of them serving the application.
```
Comment (tfpauly) [inline draft-ietf-taps-interface.md:693]:
```suggestion
If more than one Remote Endpoint is specified on the Preconnection, the
application is indicating that it expects all of the Remote Endpoints to
offer an equivalent service, and that the Transport Services system can choose
any of them for a Connection.
```

**Review by tfpauly — CHANGES_REQUESTED** [review]

**Review by gorryfair — APPROVED** [review]

**Review by tfpauly — APPROVED** [review]

**Pull request merged**